src/pathfinding.cpp

```diff
@@ -174,13 +218,20 @@ std::vector<tripoint> map::route( const tripoint &f, const tripoint &t,
     clip_to_bounds( maxx, maxy, maxz );
 
     pathfinder pf( minx, miny, maxx, maxy );
-    pf.add_point( 0, 0, f, f );
     // Make NPCs not want to path through player
     // But don't make player pathing stop working
-    if( f != pl_pos && t != pl_pos ) {
-        pf.close_point( pl_pos );
+    for( const auto &p : pre_closed ) {
+        if( p.x >= minx && p.x < maxx && p.y >= miny & p.y < maxy ) {
```

Comment:
There is an unary & breaking the builds (except for Linux 32 bit) because of the warning associated with it.